ts/src/main/scala/ch/epfl/ts/brokers/ExampleBroker.scala

```diff
@@ -0,0 +1,129 @@
+        println("Duplicate Id")
+        //TODO(sygi): reply to the trader that registration failed
+        //TODO(sygi): return?
+      }
+      mapping = mapping + (id -> sender())
+      context.actorOf(Props[Wallet], "wallet" + id)
+      sender() ! ConfirmRegistration
+    }
+    case FundWallet(uid, curr, value) => {
+      log.debug("Broker: got a request to fund a wallet")
+      val replyTo = sender
+      executeForWallet(uid, FundWallet(uid, curr, value), {
+        case WalletConfirm(uid) => {
+          println("Broker: Wallet confirmed")
+          replyTo ! WalletConfirm(uid)
+        }
+        case WalletInsufficient(uid) => {
+          println("Broker: insufficient funds")
+          replyTo ! WalletInsufficient(uid)
+        }
+        case p => println("FundWallet: Unexpected message " + p)
+      })
+    }
+    case GetWalletFunds(uid) => { //TODO(sygi): check if trader asks for his wallet
+      log.debug("Broker: got a get show wallet request")
+      val replyTo = sender
+      executeForWallet(uid, GetWalletFunds(uid), {
+        case w: WalletFunds => {
+          replyTo ! w
+        }
+      })
+    }
+
+    //TODO(sygi): refactor charging the wallet/placing an order
+    //TODO(sygi): other orders
+    case o: MarketBidOrder => {
+      log.debug("Broker: received order")
+      val replyTo = sender
+      val uid = o.chargedTraderId()
+      val placementCost = o.costValue()
+      val costCurrency = o.costCurrency()
+      executeForWallet(uid, FundWallet(uid, costCurrency, -placementCost), {
+        case WalletConfirm(uid) => {
+          send(o)
+          replyTo ! WalletConfirm(uid) //means: order placed
+          log.debug("Broker: Wallet confirmed")
+        }
+        case WalletInsufficient(uid) => {
+          replyTo ! WalletInsufficient(uid)
+          log.debug("Broker: insufficient funds")
+        }
+        case _ => log.debug("Unexpected message")
+      })
+    }
+
+    case Transaction(mid, price, volume, timestamp, whatC, withC, buyerId, buyOrderId, sellerId, sellOrderId) => {
+      log.debug("Broker: received transaction: " + buyerId + " " + sellerId)
+      if (mapping.contains(buyerId)){
+        //TODO(sygi): refactor charging wallet
+        val replyTo = mapping.getOrElse(buyerId, null)
+        executeForWallet(buyerId, FundWallet(buyerId, whatC, volume * price), { //TODO(sygi): times or divided by price?
+          case WalletConfirm(uid) => {
+            log.debug("Broker: Transaction executed")
+            replyTo ! WalletConfirm(uid) //TODO(sygi): change to some better information (or don't inform at all, as everybody gets Transaction)
+          }
+          case p => log.debug("Broker: A wallet replied with an unexpected message: " + p)
+        })
+      }
+      //TODO(sygi): do the same with the seller
+    }
+    case p => println("Broker: received unknown " + p)
+  }
+
+  def executeForWallet(uid: Long, question: WalletState, f: PartialFunction[Any, Unit]) = {
```

Comment:
Is `f` a callback? Then, following a JavaScript convention, could you call it `cb`?